src/main/java/dp/handler/Handler.java

```diff
@@ -84,9 +86,23 @@ public void listen(final ExportedFile message) {
 
         } catch (final IOException e) {
             LOGGER.error("error when exporting filter {}. exception : {}", message.getFilterId(), e);
+        } finally {
+            LOGGER.info("cleaning up workbook resources");
+            closeWorkbook((SXSSFWorkbook) workbook);
         }
+        LOGGER.info("completed processing kafka message", message.getFilterId());
+    }
 
-        LOGGER.info("exported completed for filterID: {}", message.getFilterId());
+    private void closeWorkbook(SXSSFWorkbook workbook) {
+        if (workbook != null) {
+            try {
+                // you must call dispose on SXSSFWorkbook to discard the generated temp files.
+                workbook.dispose();
+                workbook.close();
+            } catch (IOException e) {
+                LOGGER.error("unexpected error while attempting to close workbook, {}", e.getMessage());
```

Comment:
Its better to log the error like this `LOGGER.error("unexpected error while attempting to close workbook", e);` instead of calling e.getMessage(). This should create a full stack trace instead of just the exception message.